src/cocoa/toga_cocoa/constraints.py

```diff
@@ -35,39 +35,46 @@ def container(self):
 
     @container.setter
     def container(self, value):
-        self._container = value
-        # print("Add constraints for", self.widget, 'in', self.container, self.widget.interface.layout)
-        self.left_constraint = NSLayoutConstraint.constraintWithItem_attribute_relatedBy_toItem_attribute_multiplier_constant_(
-            self.widget.native, NSLayoutAttributeLeft,
-            NSLayoutRelationEqual,
-            self.container.native, NSLayoutAttributeLeft,
-            1.0, 10  # Use a dummy, non-zero value for now
-        )
-        self.container.native.addConstraint(self.left_constraint)
+        if value is None and self.container:
+            self.container.native.removeConstraint(self.width_constraint)
+            self.container.native.removeConstraint(self.height_constraint)
+            self.container.native.removeConstraint(self.left_constraint)
+            self.container.native.removeConstraint(self.top_constraint)
+            self._container = value
+        else:
+            self._container = value
+            # print("Add constraints for", self.widget, 'in', self.container, self.widget.interface.layout)
+            self.left_constraint = NSLayoutConstraint.constraintWithItem_attribute_relatedBy_toItem_attribute_multiplier_constant_(
```

Comment:
Unfortunately there's no way to make flake8 happy on this line; this is a case where the only option is to add `# NOQA:E501` to the line.

As a larger project, it would be worth exploring adding an aliasing mechanism, or some other way to disambiguate arguments (like `attribute` in this case). Long method names like these definitely affect the readability of Python code (and, as you've found, impacts onto flake8 checks). I've opened beeware/rubicon-objc#148 to discuss options.